### PR TITLE
Add samples dir as input to asciidoc and test

### PR DESCRIPTION
**Pull request by wolfs**
Context: https://github.com/gradle-guides/using-build-cache/pull/22#discussion_r130186481

### CODE REVIEW

**Comment by wolfs** [comment]
@bmuschko I added some tests. Please have a look.

File: src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy [hunk 1]
```diff
@@ -89,6 +91,42 @@ include::{samplesoutputdir}/helloWorld/build.out[]
...
+        def samplesOutputDir = createSamplesOutputDir()
+
+        when:
+        def result = build(asciiDoctorTask)
+
+        then:
+        result.task(asciiDoctorTask).outcome == TaskOutcome.SUCCESS
+
+        when:
+        new File(samplesCodeDir, "build.gradle") << """apply plugin: java"""
```
Comment (bmuschko) [inline src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy]: No need to for multi-line String.

File: src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy [hunk 2]
```diff
@@ -89,6 +91,42 @@ include::{samplesoutputdir}/helloWorld/build.out[]
         htmlFile.contains("Unresolved directive in index.adoc - include::${temporaryFolder.root.canonicalPath}/samples/output/helloWorld/build.out[]")
     }
 
+    def "asciidoctor is out of date if samples change"() {
+        given:
+        def asciiDoctorTask = ":asciidoctor"
+        def contentsDir = createContentsDir()
+        new File(contentsDir, "index.adoc") << """This is some sample ascii source"""
```
Comment (bmuschko) [inline src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy]: No need to for multi-line String.

File: src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy [hunk 3]
```diff
@@ -89,6 +91,42 @@ include::{samplesoutputdir}/helloWorld/build.out[]
...
+
+        when:
+        new File(samplesCodeDir, "build.gradle") << """apply plugin: java"""
+        result = build(asciiDoctorTask)
+
+        then:
+        result.task(asciiDoctorTask).outcome == TaskOutcome.SUCCESS
+
+        when:
+        new File(samplesOutputDir, "my-task-output.log") << """Build SUCCESSFUL"""
```
Comment (bmuschko) [inline src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy]: No need to for multi-line String.

File: src/functTest/groovy/org/gradle/guides/TestJvmCodePluginFunctionalTest.groovy [hunk 4]
```diff
@@ -97,4 +99,46 @@ class MyTest extends Specification {
...
         noExceptionThrown()
     }
+
+    def "test is out of date if samples change"() {
+        def testDir = temporaryFolder.newFolder('src', 'test', 'groovy')
+        new File(testDir, 'MyTest.groovy') << """
+import spock.lang.Specification
+
+class MyTest extends Specification {
+    def "can resolve samplesDir system property"() {
```
Comment (bmuschko) [inline src/functTest/groovy/org/gradle/guides/TestJvmCodePluginFunctionalTest.groovy]: Test method name should have a better name. Looks like copy/paste from a different one.

File: src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy (hunk 2 above)
Comment (wolfs) [inline src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy]: fixed.

File: src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy (hunk 1 above)
Comment (wolfs) [inline src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy]: fixed.

File: src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy (hunk 3 above)
Comment (wolfs) [inline src/functTest/groovy/org/gradle/guides/BasePluginFunctionalTest.groovy]: fixed.

File: src/functTest/groovy/org/gradle/guides/TestJvmCodePluginFunctionalTest.groovy (hunk 4 above)
Comment (wolfs) [inline src/functTest/groovy/org/gradle/guides/TestJvmCodePluginFunctionalTest.groovy]: changed name.

**Comment by wolfs** [comment]
@bmuschko I addressed your comments. Please have another look.

**Pull request merged**